helper/inet/inet_test.go

```diff
@@ -0,0 +1,37 @@
+package inet
+
+import (
+	"reflect"
+	"testing"
+)
+
+func TestAdvertiseFromSubnet(t *testing.T) {
+	// CIDR is out of range
+	if ad, err := AdvertiseIpFromSubnet("127.0.0.0/43:4648"); err == nil {
```

Comment:
You are testing different invalid and valid cases. Each of the cases should be in its own test function.